server/src/internalClusterTest/java/org/opensearch/search/SearchCancellationIT.java

```diff
@@ -207,10 +207,7 @@ private void verifyCancellationException(ShardSearchFailure[] failures) {
             // failure may happen while executing the search or while sending shard request for next phase.
             // Below assertion is handling both the cases
             final Throwable topFailureCause = searchFailure.getCause();
-            assertTrue(
-                searchFailure.toString(),
-                topFailureCause instanceof TransportException || topFailureCause instanceof TaskCancelledException
-            );
+            assertTrue(searchFailure.toString(), topFailureCause instanceof RuntimeException);
```

Comment:
In the concurrent search case this is going to throw `RuntimeException` from Lucene -- https://github.com/apache/lucene/blob/branch_9_7/lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java#L724-L732